node/pkg/common/symmetric_test.go

```diff
@@ -1,24 +1,60 @@
 package common
 
 import (
+	"encoding/hex"
+	"github.com/stretchr/testify/assert"
 	"testing"
 )
 
 func TestAESGCM(t *testing.T) {
-	data := []byte("cat")
-	key := []byte("01234567890123456789012345678901")
 
-	enc, err := EncryptAESGCM(data, key)
-	if err != nil {
-		t.Fatal(err)
+	type test struct {
+		data []byte
+		enc  string
 	}
 
-	dec, err := DecryptAESGCM(enc, key)
-	if err != nil {
-		t.Fatal(err)
+	key := []byte("01234567890123456789012345678901")
+
+	tests := []test{
+		{data: []byte("lower"), enc: "7b3023dde90ca9598eff203d92145d0c363c61743edf23110bee39c1b01200fc6f"},
+		{data: []byte("UPPER"), enc: "f70a6549fd8a004551f2f3aeebc74d03efd3520aace6bd51f1b38cac8a94aae352"},
+		{data: []byte("Mixed"), enc: "95af0109040796bda46a223acebca97305aff135bd628ce70812d59a13c1a821ac"},
+		{data: []byte("AlphaNum1"), enc: "6385f1d2ae006fc7fa2b5e0b74f12b71bf25e2ec3d9ee70baf142703a7c187de08a22806d3"},
+		{data: []byte("12345"), enc: "bd083e01ca72af788a04866ddd9c0d061f64bc0a0e0d23432fbf1d6fcc724c126c"},
 	}
 
-	if string(dec) != string(data) {
-		t.Fatalf("expected %s got %s", string(data), string(dec))
+	for _, testCase := range tests {
+		// Verify that we can encrypt plain text
+		t.Run(string(testCase.data), func(t *testing.T) {
+			enc, err := EncryptAESGCM(testCase.data, key)
+			assert.Nil(t, err)
+			assert.NotNil(t, enc)
+			// AESGCM is non-deterministic, so we cannot expect consistent cipher text
```

Comment:
The reason this is non-deterministic, is that the nonce is generated on the fly and not exposed as an interface.